### PR TITLE
Adds unit test on Bind extension for the scheduler

### LINKED ISSUE #80272: Add unit tests for Scheduler framework
**What would you like to be added**:
Add unit tests for Scheduler [framework](https://github.com/congliu01/kubernetes/blob/73704f1b8e04c573185f230019cc1891375c283b/pkg/scheduler/framework/v1alpha1/framework.go#L264). 

Currently there are only [integration tests](https://github.com/congliu01/kubernetes/blob/c54c4d1962d84b0659644f8ae9c9976b547d021f/test/integration/scheduler/framework_test.go#L435-L434) for the framework. 

**Why is this needed**:
* Unit tests are easier to write and maintain. 

The integration tests requires spawning up etcd and the scheduler reconcile loop, which includes way more logic than the framework itself. It's also hard to control and inspect the input/output easily due to the fact that the Scheduler is a "black box" during integration test. As a result, we can find tests that are really hard to understand and maintain, see [example](https://github.com/kubernetes/kubernetes/issues/80227)

* Unit tests can run fast and deterministically.

Due to the asynchronous and "black box" nature of running the Scheduler reconcile loop, the current integration tests takes longer time and flaky. Here is a recent [example](https://github.com/kubernetes/kubernetes/issues/80255) of this.

* Unit tests can potentially cover more cases than integration tests.

Again, it's hard to control and inspect the system state running integration test. For example, it's not possible to inspect the full output of the Score plugin with integration test.

### PR DESCRIPTION
**Pull request by somtochiama**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
It adds more unit test for the scheduler

**Which issue(s) this PR fixes**:

Part of unit tests for #80272 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 1]
```diff
@@ -848,6 +848,162 @@ func TestFilterPlugins(t *testing.T) {
 	}
 }
 
+func TestBindPlugin(t *testing.T) {
```
Comment (haosdent) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: Should it be `TestBindPlugins`, refer to `TestFilterPlugins`.

Comment (SomtochiAma) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: Yes, it should. I will make the change

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 2]
```diff
@@ -848,6 +848,162 @@ func TestFilterPlugins(t *testing.T) {
...
+				{
+					name: "TestPlugin",
+					inj: injectedResult{BindStatus: int(Unschedulable)},
+				},
+				{
+					name: "TestPlugin 1",
+					inj: injectedResult{BindStatus: int(Success)},
+				},
+			},
+			wantStatus: NewStatus(Error, `bind plugin "TestPlugin" failed to bind pod "/": injected status`),
```
Comment (haosdent) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: Compare to `TestFilterPlugins`, seems don't have cases that cover

```
Success,Success
Error,Error
Success,Unschedulable
```

Should we follow `TestFilterPlugins` and add it as well?

Comment (SomtochiAma) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: There is no harm in more test cases. I will add them

**Comment by alculquicondor** [comment]
/hold

I'm adding a similar test in #87430, but more test cases are always welcomed.

**Comment by alculquicondor** [comment]
Could you remove the word "Fixes" from your PR description?

You could say "Part of" instead.

**Comment by SomtochiAma** [comment]
> Could you remove the word "Fixes" from your PR description?
> 
> You could say "Part of" instead.
Done. Is it okay now?

**Comment by alculquicondor** [comment]
Yeah.

Now you could rebase on top of #87430 or wait for it to be merged first.

**Comment by SomtochiAma** [comment]
#87430 has been merged and I have rebased on top of master

**Comment by alculquicondor** [comment]
Having `TestRunBindPlugins` and `TestBindPlugins` adds confusion. Could you try to enhance `TestRunBindPlugins` instead?

**Comment by SomtochiAma** [comment]
Alright

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 3]
```diff
@@ -1064,6 +1064,66 @@ func TestRunBindPlugins(t *testing.T) {
 			injects:    []Code{Skip, Success, Error},
 			wantStatus: Success,
 		},
+		{
+			name:       "NoBindPlugin",
```
Comment (alculquicondor) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: "no bind plugin, runs DefaultBinder"

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 4]
```diff
@@ -1064,6 +1064,66 @@ func TestRunBindPlugins(t *testing.T) {
...
+			injects:    []Code{},
+			wantStatus: Success,
+		},
+		{
+			name: "UnshedulableBindPlugin",
```
Comment (alculquicondor) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: "invalid status"

File: pkg/scheduler/framework/v1alpha1/framework_test.go [hunk 5]
```diff
@@ -1064,6 +1064,66 @@ func TestRunBindPlugins(t *testing.T) {
...
+			injects: []Code{Success, Error},
+			wantStatus: Success,
+		},
+		{
+			name: "SuccessUnschedulableBindPlugin",
+			injects: []Code{Success, Error},
+			wantStatus: Success,
+		},
+		{
+			name: "ErrorSuccessBindPlugin",
```
Comment (alculquicondor) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: in general, put in the name what should happen. But also, some tests seem redundant from the ones we already have.

Comment (SomtochiAma) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: Okay. I have made some changes, can you please take another look?

**Comment by alculquicondor** [comment]
/approve

Please squash. But also make sure to run `./hack/update-gofmt.sh`

**Comment by alculquicondor** [comment]
Please squash

**Comment by SomtochiAma** [comment]
I have squashed the commits

**Pull request merged**